File: talleres/taller5/main.py
def trabajo_minimo(demandas):
    # Primero creo las variables trabajo donde voy a almacenar el trabajo que se realiza en cada paso
    # y la variable carga donde guardo la cantidad de vino que estoy transportando en cada paso
    trabajo = 0
    carga = 0
    # Recorro las casas y en cada casa si quieren vender sumo ese vino a la carga y si quieren vender lo resto
    # A la variable trabajo le sumo la cantidad de vino que estoy cargando (ya que para moverlo a la casa adyacente es lo que cuesta)
    for i in range(len(demandas)):
        carga = carga + demandas[i]
        trabajo = trabajo + abs(carga)
    print(trabajo)
            
            
if __name__ == '__main__':
    cantCasas = int(input())
    while(cantCasas != 0):
        demandas = input().split()
        for i in range(len(demandas)):
            demandas[i] = int(demandas[i])
        
        trabajo_minimo(demandas)
        cantCasas = int(input())

File: talleres/taller5/Main.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        int cant_casas = scanner.nextInt();

        while(cant_casas != 0){
            int[] demandas = new int[cant_casas];
            for(int i = 0; i < cant_casas; i++){
                int demanda = scanner.nextInt();
                demandas[i] = demanda;
            }
            
            trabajoMinimo(demandas);
            cant_casas = scanner.nextInt();
        }
        scanner.close();
    }

    public static void trabajoMinimo(int[] demandas){
        int trabajo = 0;
        for(int i = 0; i < demandas.length; i++){
            if(i < demandas.length - 1){
                boolean hayMas = true;
                while(demandas[i] > 0 && hayMas){
                    for(int j = i+1; j < demandas.length; j++){
                        if(demandas[j] < 0){
                            trabajo += j - i;
                            demandas[i] -= 1;
                            demandas[j] += 1;
                        }
                    }
                    hayMas = false;
                }
            }

            if(i > 0){
                int j = i -1;
                while(demandas[i] > 0){
                    while(demandas[j] >= 0 && j > 0){
                        j--;
                    }
                    trabajo += i - j;
                    demandas[i] -= 1;
                    demandas[j] += 1;
                }

            }
        }

        System.out.println(trabajo);
    }
}
